src/read/pe/export.rs

```diff
@@ -0,0 +1,149 @@
+                    .wrapping_sub(export_va) as usize,
+                export_dir.number_of_functions.get(LE) as usize,
+            )
+            .read_error("Invalid PE export address table")?;
+        let number = export_dir.number_of_names.get(LE) as usize;
+        let names = export_data
+            .read_slice_at::<U32Bytes<_>>(
+                export_dir.address_of_names.get(LE).wrapping_sub(export_va) as usize,
+                number,
+            )
+            .read_error("Invalid PE export name table")?;
+        let base_ordinal = export_dir.base.get(LE);
+        let ordinals = export_data
+            .read_slice_at::<U16Bytes<_>>(
+                export_dir
+                    .address_of_name_ordinals
+                    .get(LE)
+                    .wrapping_sub(export_va) as usize,
+                number,
+            )
+            .read_error("Invalid PE export ordinal table")?;
+
+        // First, let's list all exports...
+        let mut exports = Vec::new();
+        for (i, address) in addresses.iter().enumerate() {
+            // Convert from an array index to an ordinal
+            // The MSDN documentation is wrong here, see https://stackoverflow.com/a/40001778/721832
+            let ordinal: u32 = match i.try_into() {
+                Err(_err) => continue,
+                Ok(index) => index,
+            };
+            let ordinal = ordinal + base_ordinal;
+            let address = address.get(LE);
+
+            // is it a regular or forwarded export?
+            if address < export_va || (address - export_va) >= export_size {
+                exports.push(Export {
+                    ordinal: ordinal,
+                    target: Target::Local(self.common.image_base.wrapping_add(address as u64)),
+                    name: None, // might be populated later
+                });
+            } else {
+                let forwarded_to = export_data
+                    .read_string_at(address.wrapping_sub(export_va) as usize)
+                    .read_error("Invalid target for PE forwarded export")?;
+                exports.push(Export {
+                    ordinal: ordinal,
+                    target: Target::Forwarded(forwarded_to),
+                    name: None, // might be populated later
+                });
+            }
+        }
+
+        // Now, check whether some (or all) of them have an associated name
+        for (name_ptr, ordinal_index) in names.iter().zip(ordinals.iter()) {
+            // Items in the ordinal array are biased.
+            // The MSDN documentation is wrong regarding this bias, see https://stackoverflow.com/a/40001778/721832
+            let ordinal_index = ordinal_index.get(LE) as u32;
+
+            let name = export_data
+                .read_string_at(name_ptr.get(LE).wrapping_sub(export_va) as usize)
+                .read_error("Invalid PE export name entry")?;
+
+            exports
+                .get_mut(ordinal_index as usize)
```

Comment:
We used to return an error if this failed. Is this change intentional?

Reply:
It is not.

However, it does not seem this would be an un-recoverable error, so I thought it was not worth throwing every export because one ordinal is wrong. In such a case, that would still return the full list of exports, but the entries with a wrong ordinal/name mapping would appear as "exports by ordinal" (and not "by name"), which seems acceptable IMO, but I can revert to the previous behaviour if you prefer.

Reply:
In general this crate tries not to ignore errors. Please revert for now at least. If it becomes problematic we can reconsider.

Reply:
OK. I've pushed a fix